tests/haskell_cabal_indirect_link/lib.cabal

```diff
@@ -0,0 +1,18 @@
+cabal-version: >=1.10
+name: lib
+version: 0.1.0.0
+build-type: Simple
+
+library
+  exposed-modules: Lib
+  build-depends: base
+  extra-libraries: c-lib c-lib2
```

Comment:
Shouldn't this only list `c-lib`? Otherwise, `c-lib2` is also a direct dependency.
(Same below).

Reply:
Let's see: `c-lib` is underlinked. If we only specify `c-lib`, Cabal is going to specify `-lc-lib` when linking. Therefore, `c-lib2` would never be considered to be linked. By specifying both `c-lib` and `c-lib2`, the linker is made aware of both libraries.

I actually ran this test without the commit that introduces `-u` and verified that it fails with errors about missing symbols as we expect when using `-dead_strip_dyld`.

Reply:
Ah, yes, you're right. Forgot that we wanted to list the transitive C libs directly because of underlinking. Sorry for the noise.